src/mono/mono/metadata/native-library.c

```diff
@@ -1240,8 +1298,17 @@ lookup_pinvoke_call_impl (MonoMethod *method, MonoLookupPInvokeStatus *status_ou
 #ifndef HOST_WIN32
 retry_with_libcoreclr:
 #endif
-	// FIXME: these flags are not getting passed correctly
-	module = netcore_lookup_native_library (alc, image, new_scope, 0);
+	cinfo = mono_custom_attrs_from_method_checked (method, error);
+	flags = get_dllimportsearchpath_flags (cinfo);
+
+	if (flags < 0) {
+		cinfo = mono_custom_attrs_from_module (m_class_get_image (method->klass), error);
```

Comment:
I think for `[assembly:Foo]` custom attributes you want `mono_custom_attrs_from_assembly.

`mono_custom_attrs_from_module` is for `[module:Foo]` .netmodule custom attributes (which aren't a thing in net5)

Reply:
`mono_custom_attrs_from_assembly` does almost exactly what `mono_custom_attrs_from_module` does, AFAICT (the first one just uses `assembly->image` and has the same code with some flags possibly different).   I chose the `_module` one because I don't know how to get a MonoAssembly from a MonoMethod.

Reply:
it doesn't.  It uses a `MONO_CUSTOM_ATTR_ASSEMBLY` token, while `module` uses a `MONO_CUSTOM_ATTR_MODULE` token.

In general `MonoImage*` and `MonoAssembly*` are not interchangeable.  an image might not be the main image of an assembly (precisely in a situation where an assembly is composed of multiple .netmodules).

You want `m_class_get_image (method->klass)->assembly`

Reply:
Ooh! I misread the `MONO_CUSTOM_ATTR_ASSEMBLY`/`MONO_CUSTOM_ATTR_MODULE` as the same string. Thanks!